### PR TITLE
Control button order issue

### LINKED ISSUE #26: JSX RichTextEditor and Control Button Order Matters Within RichTextContainer
The JSX order of the `RichTextEditor` compared to certain control buttons matters within the `RichTextContainer` which creates for some confusing behavior/bugs.

For example,
```
<RichTextContainer>
  <ControlButtonWithUseLink />
  <RichTextEditor
    initialHTML={myHtml}
  />
</RichTextContainer>
```
works as expected.  In this case, the control button with useLink setups up the addNewHTMLListener (i.e. richTextContext.addNewHTMLListener(newHtml)) and _then_ the `initialHTML` gets set on the `RichTextEditor` which causes the `fireNewHtml` function to be called, processing the links as intended.

However,
```
<RichTextContainer>
  <RichTextEditor
    initialHTML={myHtml}
  />
  <ControlButtonWithUseLink />
</RichTextContainer>
```
does not work as expected.  In this case, the `RichTextEditor`'s `initialHTML` gets set before the useLink has had a chance to `addNewHTMLListener`s... which means the links don't get processed as expected since the `fireNewHTML` function is called before the listeners are listening.

One potential solution would be to change the `useLink`'s `useNewHtmlHandler` function (and any other custom hooks that rely on utilizing `addNewHTMLListener`s) to use the `useLayoutEffect` hook instead of the `useEffect` hook. As in...
```
function useNewHtmlHandlerLayout() {
  useLayoutEffect(() => {
    richTextContext.addNewHTMLListener(newHtml)
    return () => richTextContext.removeNewHTMLListener(newHtml)

    function newHtml() {
      if (richTextContext.getContentEditableElement()) {
        const anchorElements = richTextContext.getContentEditableElement().querySelectorAll('a')
        anchorElements.forEach(processAnchorElement)
      }
    }
  }, [processAnchorElement])
}
```

In this case, the `useNewHtmlHandler` function(s) would get synchronously called before the `RichTextEditor` component calls the `useEffect` for processing the `initialHTML` and subsequently calls `fireNewHTML`... aka...
```
useEffect(() => {
    if (props.initialHTML) {
      divRef.current.innerHTML = props.initialHTML
      richTextContext.fireNewHTML()
    }
  }, [])
```

### PR DESCRIPTION
**Pull request by jairuzu**
Moves calling `fireNewHTML` for `initialHTML` out of the RichTextEditor and into the RichTextContainer in order to ensure that it gets called after 'newHTMLListeners' have been set up by custom hooks irregardless of where those hooks are used in components within the RichTextContainer.

Fixes #26

### CODE REVIEW

**Review by dvnrsn — APPROVED** [review]

File: src/rich-text-container.component.js [hunk 1]
```diff
@@ -9,6 +9,14 @@ export function RichTextContainer(props) {
   const newHTMLListenersRef = useRef([])
   const serializers = useRef([])
 
+  useEffect(() => {
+    const contentEditableElement = contextValue.getContentEditableElement();
+
+    if (contentEditableElement && contentEditableElement.innerHTML) {
+      contextValue.fireNewHTML();
+    }
+  }, [contextValue])
```
Comment (internettrans) [inline src/rich-text-container.component.js:18]: Is the reason why this fixes things that RichTextContainer is that the parent component for all the control buttons and the editor, so by the time the RichTextContainer is mounted all the other components are already mounted (regardless of order)?

Comment (jairuzu) [reply src/rich-text-container.component.js:18]: Yeah, that is my understanding and the premise for the fix.  As I tested things it seemed that child component useEffects run before parent component useEffects based on how React works through the component tree.

Comment (internettrans) [reply src/rich-text-container.component.js:18]: 👍

Comment (dvnrsn) [reply src/rich-text-container.component.js:18]: That is my understanding. It seems to me a react component runs all the UEs of children before parent. https://codesandbox.io/s/ypj17kzwxx?fontsize=14

**Pull request merged**